internal/schemautil/service.go

```diff
@@ -636,12 +636,17 @@ func copyServicePropertiesFromAPIResponseToTerraform(
 		}
 	}
 
-	userConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
+	newUserConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
 	if err != nil {
 		return err
 	}
 
-	if err := d.Set(serviceType+"_user_config", NormalizeIpFilter(d.Get(serviceType+"_user_config"), userConfig)); err != nil {
+	// Mutates user config inplace
+	oldUserConfig := d.Get(serviceType + "_user_config")
+	copySensitiveFields(oldUserConfig, newUserConfig)
```

Comment:
could we please have some kind of a wrapper which would call both of these functions, e.g. `mutateUserConfig`

Reply:
this function is too long as it is already, and moving those call into a wrapper, would actually reduce complexity since the reader shouldn't care what exactly is being mutated under the hood, rather it should be visible that the data is being changed